src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -519,10 +519,20 @@ public SymbolInfo GetSymbolInfo(ExpressionSyntax expression, CancellationToken c
                 // Named arguments handled in special way.
                 return this.GetNamedArgumentSymbolInfo((IdentifierNameSyntax)expression, cancellationToken);
             }
-            else
+            else if (expression.Parent?.Kind() == SyntaxKind.Argument && 
```

Comment:
Rather than make a special case for this construct, I suggest you add a BoundTypeExpression to the declaration, so that the type would be found through the usual mechanism.